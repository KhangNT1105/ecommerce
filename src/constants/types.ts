export type MarketType = {
  countryCode: string;
  countryName: string;
  currencyCode: string;
  currencySymbol: string;
  active: boolean;
};

export type ClientType = {
  clientId: string;
  clientName: string;
  active: boolean;
};

export type PaginationType = {
  totalPages: number;
  totalItems: number;
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | undefined;
  size: number | 10;
  page: number | undefined;
};

export type MenuItem = {
  path: string;
  name: string;
  icon: string;
  children: MenuItem[];
};

export type ProductDenominationType = {
  label: string;
  price: number;
};

export interface IProductImage {
  logo: string;
  thumbnail?: string;
  portrait?: string;
  landscape?: string;
}

export interface ICategory {
  id: string;
  name: string;
  parentCategoryId?: string;
  externalCategoryId?: string;
  vendor?: string;
}

export interface IProductItem {
  id: string;
  displayName: string;
  vendor: string;
  productId?: string;
  market?: string;
  productBrand?: string;
  ghostLinkExpiryDays?: number;
  minimumAllowableExpiryDays?: number;
  valueType?: string;
  minPrice?: number;
  maxPrice?: number;
  redemptionType?: string;
  denominations?: string[];
  productDenominations?: ProductDenominationType[];
  ghostLinkInMarket?: number;
  allocated?: number;
  inQueue?: number;
  stockOnHand?: number;
  expiring?: number;
  internalName?: string;
  originalName?: string;
  description?: string;
  externalProductId?: string;
  categoryIds?: string[];
  published?: string;
  outOfStockBehaviour?: string;
  images?: IProductImage;
  brand?: {
    brandLogo?: string;
    brandName?: string;
  };
}

export interface AnyObject {
  [key: string]: any;
}

export interface StringPropertyObject {
  [key: string]: string;
}

export const SortValues = {
  ASC: 'updatedAt.asc',
  DESC: 'updatedAt.desc'
};

export const PortalType = {
  DOMAIN: 'domain',
  SMS: 'sms'
};

export type FnType = (param?: object) => void;
export type FnSubmitFormType = (event: React.FormEvent<HTMLFormElement>) => void;

export type EXPORT_TYPE = 'txt' | 'json' | 'csv' | 'xls' | 'xml';

export type PathParamsWithIDType = {
  id: string;
};
