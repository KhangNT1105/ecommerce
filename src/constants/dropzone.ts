import Configs from 'config';

export const UPLOAD_ACCEPT_FILE_TYPE: string = 'image/jpeg, image/jpg, image/png, image/svg, image/bmp, image/gif';
export const UPLOAD_ACCEPT_FILE_VIDEO_TYPE: string =
  'video/mp4, video/webm, video/wmv, video/flv, video/ogv, video/avi, video/mov';
export const FILE_DROPZONE_MIN_SIZE: number = 0;
export const FILE_DROPZONE_MAX_SIZE: number = 10048576;
export const FILE_DROPZONE_VIDEO_MAX_SIZE: number = 1000048576;
export const FILE_SIZE_IN_MB: number = 1;
export const FILE_SIZE_IN_PX: number = 1920;
export const RESOLUTION_MIN_SIZE_LANDSCAPE: number = 0;
export const RESOLUTION_MIN_SIZE_PORTRAIT: number = 0;

export const defaultDropzone = {
  apiURL: Configs.API.UPLOAD_SERVICE,
  files: [],
  autoUpload: true,
  config: {
    accept: UPLOAD_ACCEPT_FILE_TYPE,
    disabled: false,
    noClick: false,
    noKeyboard: false,
    minSize: FILE_DROPZONE_MIN_SIZE,
    maxSize: FILE_DROPZONE_MAX_SIZE
  },
  pushData: {
    error: false,
    msg: 'OK',
    infos: {
      title: 'Hero Banner Desktop and big screens',
      url: 'https://i.pinimg.com/236x/0a/8b/fb/0a8bfb162ef8a917f1e4d4bf5cd08e4c--android-apps-cover-art.jpg',
      size: '12222'
    }
  }
};

export const defaultDropzoneVideo = {
  ...defaultDropzone,
  config: {
    ...defaultDropzone.config,
    accept: UPLOAD_ACCEPT_FILE_VIDEO_TYPE,
    maxSize: FILE_DROPZONE_VIDEO_MAX_SIZE
  }
};

export const defaultDropzonePreviewInit = {
  files: [],
  onFileRemove: () => {},
  onFileUpload: () => {},
  status: false
};
