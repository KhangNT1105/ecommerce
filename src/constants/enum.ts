// TODO: All enum should be listed here
import { TypeOptions, ToastPosition } from 'react-toastify';
import { EXPORT_TYPE } from './types';

export const COLOR = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning',
  DANGER: 'danger',
  LINK: 'link'
};

export const SIZE = {
  LARGE: 'lg',
  SMALL: 'sm'
};

export const DURATION = {
  TOAST: 5000,
  TOAST_TRANSITION: 1000
};

interface POSITION_OPTIONS {
  [pos: string]: ToastPosition;
}
export const POSITION: POSITION_OPTIONS = {
  TOP_LEFT: 'top-left',
  TOP_RIGHT: 'top-right',
  TOP_CENTER: 'top-center',
  BOTTOM_LEFT: 'bottom-left',
  BOTTOM_RIGHT: 'bottom-right',
  BOTTOM_CENTER: 'bottom-center'
};

interface TYPE_OPTIONS {
  [type: string]: TypeOptions;
}
export const TOAST_TYPE: TYPE_OPTIONS = {
  SUCCESS: 'success',
  ERROR: 'error'
};

export const DATE_RANGE = {
  TODAY: 'today',
  LAST_7_DAYS: 'last7days',
  LAST_30_DAYS: 'last30days',
  LAST_WEEK: 'lastweek',
  LAST_MONTH: 'lastmonth',
  CUSTOM_DATE: 'customdate'
};

interface FILE_TYPE {
  [pos: string]: EXPORT_TYPE;
}
export const EXPORT_FILE_TYPE: FILE_TYPE = {
  CSV: 'csv',
  XLS: 'xls'
};

export const BREAKPOINTS = {
  EXTRA_LARGE: 1200,
  LARGE: 992,
  MEDIUM: 768,
  SMALL: 576
};

export const DROPZONE_EVENT_TYPE = {
  DROP: 'drop',
  CLICK: 'click'
};

export const ENTITIES = {
  PATH: {
    CONTENT: 'content',
    LICENSES: 'licenses',
    USER_MANAGEMENT: 'user-management',
    SYSTEM_CONFIGURATION: 'system-configuration'
  },
  NAME: {
    SHOWS: 'shows',
    SCHEDULES: 'schedules',
    CHANNELS: 'channels',
    CATALOGS: 'catalogs',
    CATEGORIES: 'categories',
    CASTINGS: 'casts',
    CLIPS: 'clips',
    PLAYLISTS: 'playlists',
    UPDATES: 'updates',
    CONTRACTS: 'contracts',
    LICENSORS: 'licensors',
    SYSTEM_USERS: 'system-users',
    USER_GROUPS: 'user-groups',
    SEASONS: 'seasons',
    CONFIGURATIONS: 'configuration',
    EPISODES: 'episodes',
    PAYMENT_GATEWAYS: 'payment-gateways',
    RISK_ENGINE_RULES_CONFIG: 'risk-engine'
  }
};

export const PUBLIC_PAGES = {
  NAME: {
    WELCOME: '/',
    LOGIN: 'login',
    TAC: 'term-and-conditions',
    ACCESS_DENIED: 'access-denied-403',
    DEMO: '/demo',
    PRODUCTS: 'sanpham',
    PRODUCT: 'sanpham/:id',
    CART: 'cart',
    CHECKOUT: 'checkout'
  }
};

export const LANGUAGES_CODE = {
  VN: 'vn',
  EN: 'en'
};

// This is API's entities from BE
export const API_ENTITIES_NAME = {
  SHOWS: 'shows',
  SCHEDULES: 'schedules',
  CHANNELS: 'channels',
  CATALOGS: 'catalogs',
  CATEGORIES: 'categories',
  CASTINGS: 'casts',
  CLIPS: 'clips',
  PLAYLISTS: 'playlists',
  UPDATES: 'updates',
  CONTRACTS: 'contracts',
  LICENSORS: 'licensors',
  SYSTEM_USERS: 'usergroups', // BE don't have entity's name : users | using same usergroups entity
  USER_GROUPS: 'usergroups',
  SEASONS: 'seasons',
  CONFIGURATIONS: 'configurations',
  EPISODES: 'episodes',
  RISK_ENGINES: 'riskengines'
};

export const PAGE_ACTIONS = {
  CREATE: 'create',
  EDIT: 'edit',
  DELETE: 'delete'
};

export const API_ERROR_CODE = {
  TOKEN_INVALID: '1012',
  TOKEN_EXPIRED: '1011',
  REFRESH_TOKEN_EXPIRED: '1014',
  REFRESH_TOKEN_INVALID: '1015'
};

export const PERMISSION_PREFIX = {
  VIEW: 'VIEW',
  CREATE: 'CREATE',
  EDIT: 'EDIT',
  EDIT_APPROVAL: 'EDIT_APPROVAL',
  EDIT_VIEW_CONTRACT: 'EDIT_VIEW_CONTRACT',
  ACTIVATE: 'ACTIVATE',
  DEACTIVATE: 'DEACTIVATE',
  APPROVE: 'APPROVE',
  REJECT: 'REJECT',
  DELETE: 'DELETE',
  RETRIEVE_BCM: 'RETRIEVE_BCM'
};

export const PAGE_ENTITY = {
  HOME: 'HOME'
};

export const BUTTONS_VISIBILITY = {
  SHOW: 'show'
};
export const PAGE_ACCESSIBILITY = {
  SHOW: 'show'
};

export const CLASS_NAME = {
  READONLY: 'read-only',
  DISABLED: 'disabled'
};

export const SEARCH_BOX_SUBMIT_DELAY = 600;

export const SHOWS_TYPES = {
  MOVIE: 'movie',
  SERIES: 'series',
  LIFE_EVENT: 'life event',
  THEATRICAL_PLAY: 'theatrical play',
  PROGRAM: 'program',
  MATCH: 'match',
  SEASONS: 'seasons',
  CHANNELS: 'channels',
  SCHEDULED: 'scheduled',
  SCHEDULING: 'ready for scheduling'
};

export const IMAGE_TYPES = {
  POSTER: 'poster',
  LANDSCAPE: 'landscape',
  POSTER_CLEAN: 'poster clean',
  LANDSCAPE_CLEAN: 'landscape clean',
  POSTER_WITH_LOGO: 'poster with logo',
  LANDSCAPE_WITH_LOGO: 'landscape with logo',
  LOGO_TITLE: 'logo title',
  HERO_BANNER_DESKTOP: 'hero banner desktop',
  HERO_BANNER_MOBILE: 'hero banner mobile'
};

export const INPUT_FIELD = {
  MAX_LENGTH_NORMAL: 100,
  MAX_LENGTH_URL: 1000
};

export const SORTING_DIRECTION = {
  ASCENDING: 'asc',
  DESCENDING: 'desc'
};
const SECONDS = {
  MINUTE: 60,
  HOUR: 3600,
  DAY: 86400,
  MONTH: 2592000,
  YEAR: 31536000
};
const ENUMS = {
  COLOR,
  SIZE,
  DURATION,
  POSITION,
  TOAST_TYPE,
  DATE_RANGE,
  EXPORT_FILE_TYPE,
  BREAKPOINTS,
  DROPZONE_EVENT_TYPE,
  SECONDS
};

export default ENUMS;
