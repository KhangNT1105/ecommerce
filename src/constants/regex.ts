const REGEX = {
  NUMBER_ONLY: /^[0-9]([.,][0-9])?$/,
  DIGIT_ONLY: /^[0-9]+$/
};

export default REGEX;
