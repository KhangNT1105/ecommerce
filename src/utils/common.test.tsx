import {
  checkRenderDoubleValue,
  getPrices,
  classConcat,
  checkEntityActionNonApproval,
  checkSMEntityActionNonApproval,
  checkEntityActionApproval,
  trimAllValueWithObjectField,
  renderSelectOptions,
  getAllSubGenres
} from './common';
import { ACTIVATE, ENTITY_STATUS_FROM_SERVER, ENTITY_STATUS, DEACTIVATE, DELETE, APPROVE, REJECT } from '../constants';

describe('common', () => {
  describe('checkRenderDoubleValue', () => {
    test('render default', () => {
      const result = checkRenderDoubleValue();
      expect(result).toMatchSnapshot();
    });
    test('render with value 1', () => {
      const result = checkRenderDoubleValue('abc');
      expect(result).toMatchSnapshot();
    });
    test('render with value 1 && value 2', () => {
      const result = checkRenderDoubleValue('abc', 'bcd');
      expect(result).toMatchSnapshot();
    });
    test('render with value 1 && value 2 && value 3', () => {
      const result = checkRenderDoubleValue('abc', 'bcd', '1');
      expect(result).toMatchSnapshot();
    });
  });
  describe('getPrices', () => {
    test('should return checkRenderDoubleValue when valueType is RANGE', () => {
      const productDetail = {
        valueType: 'RANGE',
        minPrice: 0,
        maxPrice: 100
      };
      const wrapper = getPrices(productDetail);
      expect(wrapper).toMatchSnapshot();
    });
    test('should return value denominations when valueType is SLAB', () => {
      const productDetail = {
        valueType: 'SLAB',
        denominations: [
          { label: 'apple', price: 500 },
          { label: 'banana', price: 400 }
        ]
      };
      const wrapper = getPrices(productDetail);
      expect(wrapper).toEqual('400, 500');
    });
    test('should return undefined when valueType is not SLAB && RANGE', () => {
      const productDetail = {
        valueType: 'A',
        denominations: [
          { label: 'apple', price: 500 },
          { label: 'banana', price: 400 }
        ]
      };
      const wrapper = getPrices(productDetail);
      expect(wrapper).toEqual(undefined);
    });
  });

  describe('classConcat', () => {
    test('should return empty string when classes is empty array', () => {
      expect(classConcat([])).toEqual('');
    });
  });

  describe('checkEntityActionNonApproval', () => {
    test('should allow activate when action is Activate and status from server is activated', () => {
      const res = checkEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.ACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow activate when action is Activate and status from server is inactive', () => {
      const res = checkEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.INACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow activate when action is Activate and status from server is draft', () => {
      const res = checkEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.DRAFT);
      expect(res).toEqual(true);
    });
    test('should allow activate when action is Activate and status from server is updated', () => {
      const res = checkEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED);
      expect(res).toEqual(true);
    });
    test('should allow deactivate when action is Deactivate and status from server is activated', () => {
      const res = checkEntityActionNonApproval(DEACTIVATE, ENTITY_STATUS_FROM_SERVER.ACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow deactivate when action is Deactivate and status from server is updated', () => {
      const res = checkEntityActionNonApproval(DEACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED);
      expect(res).toEqual(true);
    });
    test('should allow delete when action is Delete and status from server is inactive', () => {
      const res = checkEntityActionNonApproval(DELETE, ENTITY_STATUS_FROM_SERVER.INACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow delete when action is Delete and status from server is draft', () => {
      const res = checkEntityActionNonApproval(DELETE, ENTITY_STATUS_FROM_SERVER.DRAFT);
      expect(res).toEqual(true);
    });
    test('should allow approve when action is Approve and status from server is pending approval', () => {
      const res = checkEntityActionApproval(APPROVE, ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING);
      expect(res).toEqual(true);
    });
    test('should allow reject when action is Reject and status from server is pending approval', () => {
      const res = checkEntityActionApproval(REJECT, ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING);
      expect(res).toEqual(true);
    });
  });

  describe('checkSMEntityActionNonApproval', () => {
    test('should allow activate when action is Activate and status from server is inactive', () => {
      const res = checkSMEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.INACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow activate when action is Activate and status from server is draft', () => {
      const res = checkSMEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.DRAFT);
      expect(res).toEqual(true);
    });
    test('should allow activate when action is Activate and status from server is updated', () => {
      const res = checkSMEntityActionNonApproval(ACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED);
      expect(res).toEqual(true);
    });
    test('should allow deactivate when action is Deactivate and status from server is activated', () => {
      const res = checkSMEntityActionNonApproval(DEACTIVATE, ENTITY_STATUS_FROM_SERVER.ACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow deactivate when action is Deactivate and status from server is updated', () => {
      const res = checkSMEntityActionNonApproval(DEACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED);
      expect(res).toEqual(true);
    });
    test('should allow delete when action is Delete and status from server is inactive', () => {
      const res = checkSMEntityActionNonApproval(DELETE, ENTITY_STATUS_FROM_SERVER.INACTIVATE);
      expect(res).toEqual(true);
    });
    test('should allow delete when action is Delete and status from server is draft', () => {
      const res = checkSMEntityActionNonApproval(DELETE, ENTITY_STATUS_FROM_SERVER.DRAFT);
      expect(res).toEqual(true);
    });
    test('should allow approve when action is Approve and status from server is pending approval', () => {
      const res = checkSMEntityActionNonApproval(APPROVE, ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING);
      expect(res).toEqual(true);
    });
    test('should allow reject when action is Reject and status from server is pending approval', () => {
      const res = checkSMEntityActionNonApproval(REJECT, ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING);
      expect(res).toEqual(true);
    });
  });

  describe('trimAllValueWithObjectField', () => {
    test('should trim with object has string value', () => {
      const objectField = {
        name: '  test ',
        status: '  test    '
      };
      const expectedValue = {
        name: 'test',
        status: 'test'
      };
      expect(trimAllValueWithObjectField(objectField)).toEqual(expectedValue);
    });
    test('should trim with object has one nested object and string value', () => {
      const objectField = {
        name: '  test ',
        status: {
          new: ' new ',
          old: ' new '
        }
      };
      const expectedValue = {
        name: 'test',
        status: {
          new: 'new',
          old: 'new'
        }
      };
      expect(trimAllValueWithObjectField(objectField)).toEqual(expectedValue);
    });
    test('should trim with object has two nested object', () => {
      const objectField = {
        name: {
          firstName: ' test ',
          lastName: ' test '
        },
        status: {
          new: ' new ',
          old: ' old '
        }
      };
      const expectedValue = {
        name: {
          firstName: 'test',
          lastName: 'test'
        },
        status: {
          new: 'new',
          old: 'old'
        }
      };
      expect(trimAllValueWithObjectField(objectField)).toEqual(expectedValue);
    });
    test('should trim with object has two nested object with null or undefined value', () => {
      const objectField = {
        name: {
          firstName: undefined,
          lastName: ' test '
        },
        status: {
          new: null,
          old: ' old '
        }
      };
      const expectedValue = {
        name: {
          firstName: undefined,
          lastName: 'test'
        },
        status: {
          new: null,
          old: 'old'
        }
      };
      expect(trimAllValueWithObjectField(objectField)).toEqual(expectedValue);
    });
  });

  describe('renderSelectOptions', () => {
    test('should render select options correct', () => {
      const data = [
        {
          id: '1',
          value: 'minute',
          label: 'Minute'
        },
        {
          id: '2',
          value: 'hour',
          label: 'Hour'
        },
        {
          id: '3',
          value: 'day',
          label: 'Day'
        }
      ];
      const result = renderSelectOptions(data);
      expect(result.props.children[1].length).toEqual(3);
    });
  });
  describe('getAllSubGenres', () => {
    test('should return all subGenres with genreList', () => {
      expect(
        getAllSubGenres([
          {
            code: 'test1',
            id: 'test-id',
            children: [
              {
                code: 'test-subgenre',
                label: {
                  ar: 'label-ar'
                },
                id: 'subgenre-id'
              }
            ],
            label: {}
          },
          {
            code: 'test1',
            id: 'test-id',
            label: {}
          }
        ] as any)
      ).toEqual([
        {
          code: 'test-subgenre',
          label: {
            ar: 'label-ar'
          },
          id: 'subgenre-id'
        }
      ]);
    });
  });
});
