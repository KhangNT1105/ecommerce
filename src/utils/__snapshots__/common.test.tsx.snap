// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common checkRenderDoubleValue render default 1`] = `
<React.Fragment>
   
  
</React.Fragment>
`;

exports[`common checkRenderDoubleValue render with value 1 && value 2 && value 3 1`] = `
<React.Fragment>
  abc
   
  <b>
    1
  </b>
   - 
  bcd
</React.Fragment>
`;

exports[`common checkRenderDoubleValue render with value 1 && value 2 1`] = `
<React.Fragment>
  abc
   
   - 
  bcd
</React.Fragment>
`;

exports[`common checkRenderDoubleValue render with value 1 1`] = `
<React.Fragment>
  abc
   
  
</React.Fragment>
`;

exports[`common getPrices should return checkRenderDoubleValue when valueType is RANGE 1`] = `
<React.Fragment>
  0
   
  
  100
</React.Fragment>
`;
